server/lib/src/idm/ldap.rs

```diff
@@ -389,74 +394,61 @@ impl LdapServer {
-                        r
-                    })
-                });
+                LdapBindTarget::ApiToken
             }
-        } else {
+        } else if dn == "dn=token" {
             // Is the passed dn requesting token auth?
             // We use dn= here since these are attr=value, and dn is a phantom so it will
             // never be present or match a real value. We also make it an ava so that clients
             // that over-zealously validate dn syntax are happy.
-            if dn == "dn=token" {
-                let lae = LdapTokenAuthEvent::from_parts(pw.to_string())?;
-                return idm_auth.token_auth_ldap(&lae, ct).await.and_then(|r| {
-                    idm_auth.commit().map(|_| {
-                        if r.is_some() {
-                            security_info!(%dn, "✅ LDAP Bind success");
-                        } else {
-                            security_info!(%dn, "❌ LDAP Bind failure");
-                        };
-                        r
-                    })
-                });
-            };
-
+            LdapBindTarget::ApiToken
+        } else {
             let rdn = self
                 .binddnre
                 .captures(dn)
                 .and_then(|caps| caps.name("val"))
                 .map(|v| v.as_str().to_string())
                 .ok_or(OperationError::NoMatchingEntries)?;
 
-            trace!(?rdn, "relative dn");
-
             if rdn.is_empty() {
                 // That's weird ...
                 return Err(OperationError::NoMatchingEntries);
             }
 
-            idm_auth.qs_read.name_to_uuid(rdn.as_str()).map_err(|e| {
+            let uuid = idm_auth.qs_read.name_to_uuid(rdn.as_str()).map_err(|e| {
                 request_error!(err = ?e, ?rdn, "Error resolving rdn to target");
                 e
-            })?
+            })?;
+
+            LdapBindTarget::Account(uuid)
         };
 
-        let lae = LdapAuthEvent::from_parts(target_uuid, pw.to_string())?;
-        idm_auth.auth_ldap(&lae, ct).await.and_then(|r| {
-            idm_auth.commit().map(|_| {
-                if r.is_some() {
-                    security_info!(%dn, "✅ LDAP Bind (password) success");
-                } else {
-                    security_info!(%dn, "❌ LDAP Bind failure");
-                };
-                r
-            })
-        })
+        let result = match target {
+            LdapBindTarget::Account(uuid) => {
+                let lae = LdapAuthEvent::from_parts(uuid, pw.to_string())?;
+                idm_auth.auth_ldap(&lae, ct).await?
+            }
+            LdapBindTarget::ApiToken => {
+                let lae = LdapTokenAuthEvent::from_parts(pw.to_string())?;
+                idm_auth.token_auth_ldap(&lae, ct).await?
+            }
+        };
+
+        idm_auth.commit()?;
+
+        if result.is_some() {
+            security_info!(%dn, "✅ LDAP Bind success");
+        } else {
+            security_info!(%dn, "❌ LDAP Bind failure");
```

Comment:
Please update the logging here so it logs `target` to show what was attempted/succeeded? DN is handy but also understanding how the system treated it would be good too.